mllib/src/main/scala/org/apache/spark/mllib/evaluation/BinaryClassificationMetrics.scala

```diff
@@ -43,10 +43,13 @@ import org.apache.spark.sql.DataFrame
 @Since("1.0.0")
 class BinaryClassificationMetrics @Since("1.3.0") (
     @Since("1.3.0") val scoreAndLabels: RDD[(Double, Double)],
-    @Since("1.3.0") val numBins: Int) extends Logging {
+    @Since("1.3.0") var numBins: Int) extends Logging {
```

Comment:
I add this method is because, in the code, there is "// TODO: Allow the user to vary the number of bins using a setBins method in BinaryClassificationMetrics. " but I find there is no setBins method. So I add it. I we not plan to add this method, maybe we can remove this comments.